system/include/compat/avxintrin.h

```diff
@@ -11,14 +11,505 @@
 #error "AVX instruction set not enabled"
 #endif
 
+#include <emmintrin.h>
+#include <emscripten/console.h>
 #include <nmmintrin.h>
+#include <pmmintrin.h>
+#include <smmintrin.h>
+#include <tmmintrin.h>
+#include <xmmintrin.h>
 
-static __inline__ __m128 __attribute__((__always_inline__, __nodebug__))
-_mm_broadcast_ss(const float *__mem_addr)
-{
-  return (__m128)wasm_v32x4_load_splat(__mem_addr);
+typedef struct {
+  __m128d v0;
+  __m128d v1;
+} __m256d;
+
+typedef struct {
+  __m128 v0;
+  __m128 v1;
+} __m256;
+
+typedef struct {
+  __m128i v0;
+  __m128i v1;
+} __m256i;
+
+typedef long long __m128i_u;
+typedef struct {
+  __m128i_u v0;
+  __m128i_u v1;
+} __m256i_u;
+
+union m256_data {
+  __m256i int_view;
+  __m256d double_view;
+  __m256 float_view;
+  __m128i_u int_u_view;
+};
+
+#define UNIMPLEMENTED(name)                                                    \
+  emscripten_err("warning: unsupported avx intrinsic: " #name "\n")
+
+/* Define the default attributes for the functions in this file. */
+#define __DEFAULT_FN_ATTRS __attribute__((__always_inline__, __nodebug__))
+
+static __inline __m256d __DEFAULT_FN_ATTRS _mm256_add_pd(__m256d __a,
+                                                         __m256d __b) {
+  __m256d ret;
+  ret.v0 = _mm_add_pd(__a.v0, __b.v0);
+  ret.v1 = _mm_add_pd(__a.v1, __b.v1);
+  return ret;
+}
```

Comment:
A bit of a silly note, though I see for other *mmintrin.h functions the visual style

```c
static __inline__ __m128d __attribute__((__always_inline__, __nodebug__))
_mm_add_pd(__m128d __a, __m128d __b)
{
  return (__m128d)wasm_f64x2_add((v128_t)__a, (v128_t)__b);
}
```
is used, which avoids use of a define `__DEFAULT_FN_ATTRS`. I would lean to having consistency, i.e. either we'd use the #define everywhere, or nowhere. I'd probably lean towards spelling the macro out, although ok to leave like this as well to avoid unnecessary review churn.